Sources/Commands/SwiftScriptTool.swift

```diff
@@ -0,0 +1,374 @@
+        abstract: "Manage and run Swift scripts",
+        discussion: "SEE ALSO: swift build, swift run, swift package, swift test",
+        version: SwiftVersion.currentVersion.completeDisplayString,
+        subcommands: [
+            Run.self,
+            Build.self,
+            Clean.self,
+            Reset.self,
+            Resolve.self,
+            List.self,
+        ],
+        helpNames: [.short, .long, .customLong("help", withSingleDash: true)])
+
+    public init() {}
+
+    public static var _errorLabel: String { "error" }
+}
+
+/// swift-run tool namespace
+extension SwiftScriptTool {
+    struct Run: ScriptCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Runs a script")
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+
+        @OptionGroup()
+        var options: ScriptToolOptions
+
+        /// Whether to print build progress.
+        @Flag(help: "Print build progress")
+        var quiet: Bool = false
+
+        func run(_ swiftTool: SwiftTool, as productName: String, at cacheDirPath: AbsolutePath) throws {
+            let output = BufferedOutputByteStream()
+            if quiet {
+                swiftTool.redirectStdoutTo(.init(output))
+            } else {
+                swiftTool.redirectStdoutToStderr()
+            }
+            // FIXME: More elegant solution?
+            print(diagnostic: .init(message: .note("Using cache: \(cacheDirPath.basename)")),
+                  stdoutStream: swiftTool.stdoutStream)
+
+            do {
+                let buildSystem = try swiftTool.createBuildSystem(explicitProduct: nil)
+                if options.shouldBuild {
+                    try buildSystem.build(subset: .product(productName))
+                }
+
+                let executablePath = try swiftTool.buildParameters().buildPath.appending(component: productName)
+                try Commands.run(executablePath,
+                        originalWorkingDirectory: swiftTool.originalWorkingDirectory,
+                        arguments: options.arguments)
+            } catch let error as ScriptError {
+                swiftTool.diagnostics.emit(error)
+                stderrStream <<< output.bytes
+                stderrStream.flush()
+                throw ExitCode.failure
+            }
+        }
+    }
+
+    struct Build: ScriptCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Prebuild a script")
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+
+        @OptionGroup()
+        var options: ScriptToolOptions
+
+        @Option(name: .shortAndLong, help: "Save the prebuilt script binary", transform: AbsolutePath.init)
+        var output: AbsolutePath?
+
+        func run(_ swiftTool: SwiftTool, as productName: String, at cacheDirPath: AbsolutePath) throws {
+            swiftTool.redirectStdoutToStderr()
```

Comment:
Is this to separate out any output from the build from the output emitted by the script, once it runs?

Reply:
Yes. This piece of code is copied from `swift-run`.